tests/test_request_message.py

```diff
@@ -1,98 +1,68 @@
-            "select * from conversation_request where initiator_id=:sender_id and receiver_id=:receiver_id",
-            {"sender_id": sender_id, "receiver_id": receiver_id},
-            one=True
-            ) is not None
+    # test sending request fail
+    def test_sending_request_fail(self, test_client, test_db):
 
-# test sending request fail
-@pytest.mark.parametrize(("sender_id", "receiver_id", "request_time", "status_code", "response_message"), (
-    (1000, 1001, 123456, 400, b'{"Error":"Bad Request.Request has already been sent"}\n'),
-    (1001, 1005, 123456, 400, b'{"Error":"Bad Request.FOREIGN KEY constraint failed"}\n')
-    ))
-def test_sending_request_fail(test_client, app, sender_id, receiver_id, request_time, status_code, response_message):
+        response = test_client.post(f"/api/request/send/1/2/123456")
+        assert response.status_code == 400
+        assert response.data == b'{"Error":"Bad Request.Request has already been sent"}\n'
 
-    response = test_client.post(f"/api/request/send/{sender_id}/{receiver_id}/{request_time}")
-    assert response.status_code == status_code
-    assert response.data == response_message
+    # test accepting request successfully
+    def test_accepting_request(self, test_client, test_db):
+        response = test_client.post("/api/request/accept/1/123456")
+        assert response.status_code == 200
+        assert response.data == b'Success'
 
-# test accepting request successfully
-def test_accepting_request(test_client, app):
-    response = test_client.post("/api/request/accept/1001/123456")
-    assert response.status_code == 200
-    assert response.data == b'Success'
 
-    with app.app_context():
-        assert query_db(
-                "select * from conversation_request where id=:request_id",
-                {"request_id": 1001},
-                one=True
-                )["accepted"] == 1
+    # test accepting request fail
+    def test_accepting_request_fail(self, test_client, test_db):
+        response = test_client.post("/api/request/accept/1002/123456")
+        assert response.status_code == 400
+        assert response.data == b'{"Error":"Bad Request.No request found"}\n'
 
-# test accepting request fail
-def test_accepting_request_fail(test_client, app):
-    response = test_client.post("/api/request/accept/1002/123456")
-    assert response.status_code == 400
-    assert response.data == b'{"Error":"Bad Request.No request found"}\n'
+    # test reject request successfully
+    def test_rejecting_request(self, test_client, test_db):
+        response = test_client.post("/api/request/reject/1")
+        assert response.status_code == 200
+        assert response.data == b'Success'
 
-# test reject request successfully
-def test_rejecting_request(test_client, app):
-    response = test_client.post("/api/request/reject/1001")
-    assert response.status_code == 200
-    assert response.data == b'Success'
+    # test rejecting request fail
```

Comment:
Why would `reject a conversation request` fail? If there is no conversation request, we can't display it in the first place right?